packages/block-library/src/navigation/style.scss

```diff
@@ -345,6 +345,13 @@
 		align-items: var(--layout-align, initial);
 	}
 
+	// If the responsive wrapper is present but overlay is not open,
+	// overlay styles shouldn't apply.
+	&:not(.is-menu-open.is-menu-open) {
+		color: inherit !important;
```

Comment:
The !important's feel appropriate enough for the way the feature works at the moment. It's possible in the future we could explore editing the contents of the modal in isolation (like template parts), at which point these would be revisited anyway.